File: javascript/leap/leap.js
const LEAP_YEAR = 4
const CENTURIAL_YEAR = 100
const LEAP_CYCLE_LENGTH = LEAP_YEAR * CENTURIAL_YEAR

export const isLeap = year => {
  return isLeapYear(year) && (isNonCenturialYear(year) || isLeapYearCycle(year))
}

function isLeapYear(year) {
  return year % LEAP_YEAR === 0
}

function isNonCenturialYear(year) {
  return !(year % CENTURIAL_YEAR === 0)
}

function isLeapYearCycle(year) {
  return year % LEAP_CYCLE_LENGTH === 0
}

File: java/leap/src/main/java/Leap.java
class Leap {
    private static final int LEAP_YEAR = 4;
    private static final int CENTURIAL_YEAR = 100;
    private static final int LEAP_CYCLE_LENGTH = LEAP_YEAR * CENTURIAL_YEAR;

    boolean isLeapYear(int year) {
        return isLeap(year) &&
            (isNonCenturialYear(year) || isLeapYearCycle(year));
    }

    private boolean isLeap(int year) {
        return year % LEAP_YEAR == 0;
    }

    private boolean isNonCenturialYear(int year) {
        return !(year % CENTURIAL_YEAR == 0);
    }

    private boolean isLeapYearCycle(int year) {
        return year % LEAP_CYCLE_LENGTH == 0;
    }
}
